python/jet/interpreter.py

```diff
@@ -43,43 +41,48 @@ def get_xir_library() -> XIRProgram:
 
         import jet
 
-        program = jet.get_xir_library()
+        program = jet.get_xir_manifest()
 
-        assert program.gates
+        assert program.declarations["gate"]
+        assert program.declarations["out"]
     """
-    lines = []
+    program = xir.Program()
 
     for name, cls in sorted(GateFactory.registry.items()):
-        # Instantiating the Gate subclass (with placeholder parameters) is an
-        # easy way to access properties such as the number of wires a Gate can
-        # be applied to.
+        # The [1:] is required because self is not explicitly passed to cls.__init__().
         param_vals = signature(cls.__init__).parameters.values()
-        # There is no need to mock parameters that have default values; the [1:]
-        # is required because self is not explicitly passed to cls.__init__().
         param_keys = [param.name for param in param_vals if param.default is param.empty][1:]
+
+        # Instantiating the Gate subclass (with placeholder parameters) is an
+        # easy way to query the number of wires a Gate acts on.
         gate = cls(*[None for _ in param_keys])
+        wires = range(gate.num_wires)
+
+        decl = xir.Declaration(type_="gate", name=name, params=param_keys, wires=wires)
+        program.add_declaration(decl)
 
-        # TODO: Replace gate definitions with gate declarations when parameter
-        #       and wire names are supported in gate declarations.
-        xir_params = "" if not param_keys else "(" + ", ".join(param_keys) + ")"
-        xir_wires = list(range(gate.num_wires))
+    for name in _get_xir_outputs():
+        decl = xir.Declaration(type_="out", name=name)
+        program.add_declaration(decl)
 
-        line = f"gate {name} {xir_params} {xir_wires}: {name} {xir_params} | {xir_wires}; end;"
-        lines.append(line)
+    return program
 
-    script = "\n".join(lines)
-    return parse_xir_script(script)
 
+def _get_xir_outputs() -> Iterator[str]:
+    """Returns the names of the supported XIR outputs."""
+    names = ((name, name.lower()) for name in ("Amplitude", "Expval", "Probabilities"))
+    yield from sorted(name for pair in names for name in pair)
 
-def run_xir_program(program: XIRProgram) -> List[Union[np.number, np.ndarray]]:
+
+def run_xir_program(program: xir.Program) -> List[Union[np.number, np.ndarray]]:
     """Executes an XIR program.
 
     Raises:
         ValueError: If the given program contains an unsupported gate or output
             statement or an invalid gate definition.
 
     Args:
-        program (xir.program.XIRProgram): XIR script to execute.
+        program (xir.Program): XIR script to execute.
```

Comment:
This isn't really a script. A `program` or a `circuit` might be more correct.
```suggestion
        program (xir.Program): XIR program to execute.
```